// @vly-ai/integrations/dist/project-thumbnail/index.js
'use strict';

// src/project-thumbnail/screenshot.ts
var CAPTURE_DELAY_MS = 1e4;
function delay(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}
function initScreenshotListener() {
  if (typeof window.html2canvas === "undefined") {
    const script = document.createElement("script");
    script.src = "https://cdn.jsdelivr.net/npm/html2canvas-pro@2.0.4/dist/html2canvas-pro.min.js";
    script.async = true;
    document.head.appendChild(script);
  }
  window.addEventListener("message", async (event) => {
    const message = event.data;
    if (message.type !== "vly-screenshot-request") return;
    try {
      if (typeof window.html2canvas === "undefined") {
        throw new Error("html2canvas not loaded yet");
      }
      const html2canvas = window.html2canvas;
      await delay(CAPTURE_DELAY_MS);
      const canvas = await html2canvas(document.body, {
        useCORS: true,
        allowTaint: false,
        backgroundColor: "#ffffff",
        logging: false,
        scale: 0.5,
        width: window.innerWidth,
        height: window.innerHeight
      });
      const format = message.format || "jpeg";
      const quality = message.quality || 0.85;
      const mimeType = `image/${format}`;
      canvas.toBlob((blob) => {
        if (!blob) {
          throw new Error("Failed to convert canvas to blob");
        }
        const reader = new FileReader();
        reader.onload = () => {
          window.parent.postMessage({
            type: "vly-screenshot-response",
            requestId: message.requestId,
            dataUrl: reader.result
          }, "*");
        };
        reader.onerror = () => {
          throw new Error("FileReader error");
        };
        reader.readAsDataURL(blob);
      }, mimeType, quality);
    } catch (error) {
      window.parent.postMessage({
        type: "vly-screenshot-error",
        requestId: message.requestId,
        error: error instanceof Error ? error.message : "Unknown error"
      }, "*");
    }
  });
}

// src/project-thumbnail/index.ts
function isVlyPlatform() {
  if (typeof window === "undefined") return false;
  const hostname = window.location.hostname;
  return hostname.endsWith(".freebuff.com") || hostname.endsWith(".vly.ai") || hostname.endsWith(".vly.sh") || hostname.endsWith(".freebuff.dev") || hostname.includes("localhost") || // For dev
  hostname.includes("127.0.0.1");
}
function initVly() {
  if (!isVlyPlatform()) {
    return;
  }
  initScreenshotListener();
}
if (typeof window !== "undefined") {
  if (document.readyState === "loading") {
    document.addEventListener("DOMContentLoaded", initVly);
  } else {
    initVly();
  }
}

exports.initScreenshotListener = initScreenshotListener;
exports.initVly = initVly;
