// @vly-ai/integrations/dist/index.js
'use strict';

var openaiCompatible = require('@ai-sdk/openai-compatible');
var ai = require('ai');

// src/ai/index.ts
var VlyAI = class {
  provider;
  config;
  constructor(config) {
    this.config = config;
    this.provider = openaiCompatible.createOpenAICompatible({
      name: "vly-gateway",
      baseURL: "https://integrations.vly.ai/v1/llm",
      headers: {
        "Authorization": `Bearer ${config.deploymentToken}`
      }
    });
  }
  getModel(modelName) {
    return this.provider(modelName || "gpt-5");
  }
  mapMessages(messages) {
    return messages.map((msg) => ({
      role: msg.role,
      content: msg.content
    }));
  }
  async completion(request, _options) {
    if (this.config.debug) {
      console.log("[Vly] Creating AI completion", { model: request.model });
    }
    try {
      const model = this.getModel(request.model);
      const messages = this.mapMessages(request.messages);
      const result = await ai.generateText({
        model,
        messages,
        temperature: request.temperature,
        maxOutputTokens: request.maxTokens
      });
      const responseData = {
        id: `vly-${Date.now()}`,
        choices: [{
          message: {
            role: "assistant",
            content: result.text
          },
          finishReason: result.finishReason || "stop"
        }],
        usage: {
          promptTokens: result.usage?.inputTokens || 0,
          completionTokens: result.usage?.outputTokens || 0,
          totalTokens: result.usage?.totalTokens || 0
        }
      };
      if (this.config.debug) {
        console.log("[Vly] AI completion successful", {
          tokensUsed: responseData.usage.totalTokens
        });
      }
      return {
        success: true,
        data: responseData
      };
    } catch (error) {
      if (this.config.debug) {
        console.error("[Vly] AI completion failed", { error: error.message });
      }
      return {
        success: false,
        error: error.message || "Request failed"
      };
    }
  }
  async streamCompletion(request, onChunk, _options) {
    if (this.config.debug) {
      console.log("[Vly] Creating streaming AI completion", { model: request.model });
    }
    try {
      const model = this.getModel(request.model);
      const messages = this.mapMessages(request.messages);
      const result = await ai.streamText({
        model,
        messages,
        temperature: request.temperature,
        maxOutputTokens: request.maxTokens
      });
      let fullResponse = "";
      for await (const delta of result.textStream) {
        fullResponse += delta;
        onChunk(delta);
      }
      const usage = await result.usage;
      const responseData = {
        id: `vly-stream-${Date.now()}`,
        choices: [{
          message: {
            role: "assistant",
            content: fullResponse
          },
          finishReason: "stop"
        }],
        usage: {
          promptTokens: usage?.inputTokens || 0,
          completionTokens: usage?.outputTokens || 0,
          totalTokens: usage?.totalTokens || 0
        }
      };
      if (this.config.debug) {
        console.log("[Vly] Streaming AI completion successful", {
          tokensUsed: responseData.usage.totalTokens
        });
      }
      return {
        success: true,
        data: responseData
      };
    } catch (error) {
      if (this.config.debug) {
        console.error("[Vly] Streaming AI completion failed", { error: error.message });
      }
      return {
        success: false,
        error: error.message || "Streaming request failed"
      };
    }
  }
  async embeddings(input, _options) {
    if (this.config.debug) {
      console.log("[Vly] Creating embeddings", { inputCount: Array.isArray(input) ? input.length : 1 });
    }
    return {
      success: false,
      error: "Embeddings not yet supported with AI SDK OpenAI-compatible provider"
    };
  }
  // Helper method to get the provider for direct AI SDK usage
  getProvider() {
    return this.provider;
  }
};

// src/client.ts
var VlyClient = class {
  config;
  constructor(config) {
    this.config = {
      deploymentToken: config.deploymentToken || "vlytomoonF2024",
      debug: config.debug ?? false
    };
  }
  async request(endpoint, method, data, options) {
    const url = `https://integrations.vly.ai${endpoint}`;
    const headers = {
      "Authorization": `Bearer ${this.config.deploymentToken}`,
      "Content-Type": "application/json",
      "X-Vly-Version": "0.1.0"
    };
    const fetchOptions = {
      method,
      headers,
      body: data ? JSON.stringify(data) : void 0
    };
    const controller = new AbortController();
    const timeout = options?.timeout || 3e4;
    const timeoutId = setTimeout(() => controller.abort(), timeout);
    fetchOptions.signal = controller.signal;
    try {
      const response = await fetch(url, fetchOptions);
      clearTimeout(timeoutId);
      const responseData = await response.json().catch(() => ({}));
      if (response.ok) {
        return {
          success: true,
          data: responseData,
          usage: responseData?.usage
        };
      } else {
        const error = responseData?.error || `Request failed with status ${response.status}`;
        if (this.config.debug) {
          console.error(`Vly API Error: ${error}`, {
            endpoint,
            status: response.status,
            data: responseData
          });
        }
        return {
          success: false,
          error
        };
      }
    } catch (error) {
      clearTimeout(timeoutId);
      const errorMessage = error.name === "AbortError" ? "Request timeout" : error.message || "Unknown error occurred";
      if (this.config.debug) {
        console.error(`Vly API Request Failed: ${errorMessage}`, {
          endpoint,
          error
        });
      }
      if (options?.retries && options.retries > 0) {
        if (this.config.debug) {
          console.log(`Retrying request... (${options.retries} retries left)`);
        }
        await new Promise((resolve) => setTimeout(resolve, 1e3));
        return this.request(endpoint, method, data, {
          ...options,
          retries: options.retries - 1
        });
      }
      return {
        success: false,
        error: errorMessage
      };
    }
  }
  log(message, data) {
    if (this.config.debug) {
      console.log(`[Vly] ${message}`, data || "");
    }
  }
};

// src/email/index.ts
var VlyEmail = class extends VlyClient {
  async send(email, options) {
    this.log("Sending email", {
      to: email.to,
      subject: email.subject
    });
    const payload = {
      to: Array.isArray(email.to) ? email.to : [email.to],
      from: email.from || "noreply@project.freebuff.dev",
      subject: email.subject,
      html: email.html,
      text: email.text,
      attachments: email.attachments,
      replyTo: email.replyTo,
      cc: email.cc ? Array.isArray(email.cc) ? email.cc : [email.cc] : void 0,
      bcc: email.bcc ? Array.isArray(email.bcc) ? email.bcc : [email.bcc] : void 0
    };
    const response = await this.request(
      "/v1/email/send",
      "POST",
      payload,
      options
    );
    if (response.success) {
      this.log("Email sent successfully", {
        id: response.data?.id,
        status: response.data?.status
      });
    }
    return response;
  }
  async sendBatch(emails, options) {
    this.log("Sending batch emails", { count: emails.length });
    const payload = {
      emails: emails.map((email) => ({
        to: Array.isArray(email.to) ? email.to : [email.to],
        from: email.from || "noreply@freebuff.com",
        subject: email.subject,
        html: email.html,
        text: email.text,
        attachments: email.attachments,
        replyTo: email.replyTo,
        cc: email.cc ? Array.isArray(email.cc) ? email.cc : [email.cc] : void 0,
        bcc: email.bcc ? Array.isArray(email.bcc) ? email.bcc : [email.bcc] : void 0
      }))
    };
    return this.request(
      "/v1/email/batch",
      "POST",
      payload,
      options
    );
  }
  async getStatus(emailId, options) {
    this.log("Getting email status", { emailId });
    return this.request(
      `/v1/email/status/${emailId}`,
      "GET",
      void 0,
      options
    );
  }
  async verifyDomain(domain, options) {
    this.log("Verifying domain", { domain });
    return this.request(
      "/v1/email/domains/verify",
      "POST",
      { domain },
      options
    );
  }
  async listDomains(options) {
    this.log("Listing email domains");
    return this.request(
      "/v1/email/domains",
      "GET",
      void 0,
      options
    );
  }
};

// src/project-thumbnail/screenshot.ts
var CAPTURE_DELAY_MS = 1e4;
function delay(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}
function initScreenshotListener() {
  if (typeof window.html2canvas === "undefined") {
    const script = document.createElement("script");
    script.src = "https://cdn.jsdelivr.net/npm/html2canvas-pro@2.0.4/dist/html2canvas-pro.min.js";
    script.async = true;
    document.head.appendChild(script);
  }
  window.addEventListener("message", async (event) => {
    const message = event.data;
    if (message.type !== "vly-screenshot-request") return;
    try {
      if (typeof window.html2canvas === "undefined") {
        throw new Error("html2canvas not loaded yet");
      }
      const html2canvas = window.html2canvas;
      await delay(CAPTURE_DELAY_MS);
      const canvas = await html2canvas(document.body, {
        useCORS: true,
        allowTaint: false,
        backgroundColor: "#ffffff",
        logging: false,
        scale: 0.5,
        width: window.innerWidth,
        height: window.innerHeight
      });
      const format = message.format || "jpeg";
      const quality = message.quality || 0.85;
      const mimeType = `image/${format}`;
      canvas.toBlob((blob) => {
        if (!blob) {
          throw new Error("Failed to convert canvas to blob");
        }
        const reader = new FileReader();
        reader.onload = () => {
          window.parent.postMessage({
            type: "vly-screenshot-response",
            requestId: message.requestId,
            dataUrl: reader.result
          }, "*");
        };
        reader.onerror = () => {
          throw new Error("FileReader error");
        };
        reader.readAsDataURL(blob);
      }, mimeType, quality);
    } catch (error) {
      window.parent.postMessage({
        type: "vly-screenshot-error",
        requestId: message.requestId,
        error: error instanceof Error ? error.message : "Unknown error"
      }, "*");
    }
  });
}

// src/project-thumbnail/index.ts
function isVlyPlatform() {
  if (typeof window === "undefined") return false;
  const hostname = window.location.hostname;
  return hostname.endsWith(".freebuff.com") || hostname.endsWith(".vly.ai") || hostname.endsWith(".vly.sh") || hostname.endsWith(".freebuff.dev") || hostname.includes("localhost") || // For dev
  hostname.includes("127.0.0.1");
}
function initVly() {
  if (!isVlyPlatform()) {
    return;
  }
  initScreenshotListener();
}
if (typeof window !== "undefined") {
  if (document.readyState === "loading") {
    document.addEventListener("DOMContentLoaded", initVly);
  } else {
    initVly();
  }
}

// src/vite-plugin.ts
function vlyPlugin() {
  return {
    name: "vly-plugin",
    transformIndexHtml: {
      order: "pre",
      handler() {
        return [
          {
            tag: "script",
            attrs: { type: "module" },
            children: `import '@vly-ai/integrations';`,
            injectTo: "head"
          },
          {
            tag: "script",
            attrs: { type: "module" },
            children: `
// Vite compile-time errors (syntax errors, failed transforms, bad imports)
if (import.meta.hot) {
  import.meta.hot.on('vite:error', (payload) => {
    window.parent.postMessage({
      type: 'vly-vite-hmr-error',
      error: {
        message: payload.err?.message || String(payload.err || payload),
        stack: payload.err?.stack,
        plugin: payload.err?.plugin,
        loc: payload.err?.loc,
      },
      timestamp: Date.now(),
    }, '*');
  });
  import.meta.hot.on('vite:beforeFullReload', () => {
    window.parent.postMessage({
      type: 'vly-vite-hmr-success',
      timestamp: Date.now(),
    }, '*');
  });
}

// Runtime errors (ReferenceError, TypeError, etc.)
window.addEventListener('error', (event) => {
  window.parent.postMessage({
    type: 'vly-vite-hmr-error',
    error: {
      message: event.message,
      stack: event.error?.stack || '',
      filename: event.filename,
      lineno: event.lineno,
      colno: event.colno,
    },
    timestamp: Date.now(),
  }, '*');
});

// Unhandled promise rejections
window.addEventListener('unhandledrejection', (event) => {
  const error = event.reason;
  window.parent.postMessage({
    type: 'vly-vite-hmr-error',
    error: {
      message: error?.message || String(error),
      stack: error?.stack || '',
    },
    timestamp: Date.now(),
  }, '*');
});`,
            injectTo: "head"
          }
        ];
      }
    }
  };
}

// src/index.ts
var DEFAULT_DEPLOYMENT_TOKEN = "vlytomoonF2024";
var VlyIntegrations = class {
  ai;
  email;
  config;
  constructor(config) {
    const normalizedToken = config.deploymentToken && config.deploymentToken.trim().length > 0 ? config.deploymentToken : DEFAULT_DEPLOYMENT_TOKEN;
    this.config = {
      deploymentToken: normalizedToken,
      debug: config.debug ?? false
    };
    this.ai = new VlyAI(this.config);
    this.email = new VlyEmail(this.config);
  }
};
function createVlyIntegrations(config) {
  return new VlyIntegrations(config);
}

exports.VlyAI = VlyAI;
exports.VlyEmail = VlyEmail;
exports.VlyIntegrations = VlyIntegrations;
exports.createVlyIntegrations = createVlyIntegrations;
exports.initScreenshotListener = initScreenshotListener;
exports.initVly = initVly;
exports.vlyPlugin = vlyPlugin;
